File: backend/src/domain/valuesobject/Quantity.go
package valuesobject

import "errors"

type Quantity struct {
	quantity int
}

func NewQuantity(quantity int) (*Quantity, error) {
	if err := validateQuantity(quantity); err != nil {
		return nil, err
	}
	return &Quantity{quantity: quantity}, nil
}

func validateQuantity(quantity int) error {
	if quantity <= 0 {
		return errors.New("quantidade nao pode ser igual a zeror ou menor")
	}
	return nil
}
func (q *Quantity) GetValue() int {
	return q.quantity
}

File: frontend/src/domain/valuesobject/Quantity.ts
export class Quantity {
  private constructor(private quantity: number) {}

  public static create(quantity: number): Quantity {
    this.validate(quantity);
    return new Quantity(quantity);
  }
  private static validate(quantity: number): void {
    if (!quantity || typeof quantity !== 'number' || quantity <= 0) {
      throw new Error('quantidade é obrigatorio e não poder ser menor que zero');
    }
  }
  public getValue(): number {
    return this.quantity;
  }
}
